src/Database/Query/SelectQuery.php

```diff
@@ -111,13 +119,6 @@ public function all(): iterable
     {
         if ($this->_results === null || $this->_dirty) {
             $this->_results = $this->execute()->fetchAll(StatementInterface::FETCH_TYPE_ASSOC);
-            if ($this->_resultDecorators) {
```

Comment:
Moving this to the statement seems good.

Reply:
Yes, `Statement` was already decorating the records using the typemap so moving user provided decorators there too makes sense, so all decorations are now in one place.